File: PolicyPlus/ExportReg.vb
Public Class ExportReg
    Dim Source As PolFile
    Public Function PresentDialog(Branch As String, Pol As PolFile, IsUser As Boolean) As DialogResult
        Source = Pol
        TextBranch.Text = Branch
        TextRoot.Text = If(IsUser, "HKEY_CURRENT_USER\", "HKEY_LOCAL_MACHINE\")
        TextReg.Text = ""
        Return ShowDialog()
    End Function
    Private Sub ButtonBrowse_Click(sender As Object, e As EventArgs) Handles ButtonBrowse.Click
        Using sfd As New SaveFileDialog
            sfd.Filter = "Registry scripts|*.reg"
            If sfd.ShowDialog = DialogResult.OK Then TextReg.Text = sfd.FileName
        End Using
    End Sub
    Private Sub ExportReg_KeyUp(sender As Object, e As KeyEventArgs) Handles Me.KeyUp
        If e.KeyCode = Keys.Escape Then DialogResult = DialogResult.Cancel
    End Sub
    Private Sub ButtonExport_Click(sender As Object, e As EventArgs) Handles ButtonExport.Click
        If TextReg.Text = "" Then
            MsgBox("Please specify a filename and path for the exported REG.", MsgBoxStyle.Exclamation)
            Exit Sub
        End If
        Dim reg As New RegFile
        reg.SetPrefix(TextRoot.Text)
        reg.SetSourceBranch(TextBranch.Text)
        Try
            Source.Apply(reg)
            reg.Save(TextReg.Text)
            MsgBox("REG exported successfully.", MsgBoxStyle.Information)
            DialogResult = DialogResult.OK
        Catch ex As Exception
            MsgBox("Failed to export REG!", MsgBoxStyle.Exclamation)
        End Try
    End Sub
End Class

File: PolicyPlus/ExportReg.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace PolicyPlus
{
    public partial class ExportReg
    {
        private PolFile Source;

        public ExportReg()
        {
            InitializeComponent();
        }
        public DialogResult PresentDialog(string Branch, PolFile Pol, bool IsUser)
        {
            Source = Pol;
            TextBranch.Text = Branch;
            TextRoot.Text = IsUser ? @"HKEY_CURRENT_USER\" : @"HKEY_LOCAL_MACHINE\";
            TextReg.Text = "";
            return ShowDialog();
        }
        private void ButtonBrowse_Click(object sender, EventArgs e)
        {
            using (var sfd = new SaveFileDialog())
            {
                sfd.Filter = "Registry scripts|*.reg";
                if (sfd.ShowDialog() == DialogResult.OK)
                    TextReg.Text = sfd.FileName;
            }
        }
        private void ExportReg_KeyUp(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
                DialogResult = DialogResult.Cancel;
        }
        private void ButtonExport_Click(object sender, EventArgs e)
        {
            if (string.IsNullOrEmpty(TextReg.Text))
            {
                Interaction.MsgBox("Please specify a filename and path for the exported REG.", MsgBoxStyle.Exclamation);
                return;
            }
            var reg = new RegFile();
            reg.SetPrefix(TextRoot.Text);
            reg.SetSourceBranch(TextBranch.Text);
            try
            {
                Source.Apply(reg);
                reg.Save(TextReg.Text);
                Interaction.MsgBox("REG exported successfully.", MsgBoxStyle.Information);
                DialogResult = DialogResult.OK;
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Failed to export REG!", MsgBoxStyle.Exclamation);
            }
        }
    }
}